### PR TITLE
guides, specifications: update for PEP 753

### PR DESCRIPTION
**Pull request by woodruffw**
This is a first blush at updating the packaging specifications for [PEP 753](https://peps.python.org/pep-0753/), now that it's been accepted.

Key changes:

1. `Home-page` and `Download-URL` have been marked as deprecated and moved into the deprecated section of the metadata spec.
2. `well-known-project-urls.rst` has been added to reflect the core spec/recommendations in PEP 753. 
3. Cross-referenced/linked between the spec above and a bunch of pages that touch on project URL handling.

(2) reflects the core spec part of the PEP, but I've tweaked the language to make it less "PEP-y" based on how the other PyPA living specs seem to be written. But I'm happy to restructure/rewrite it however others please!

Related tasks that I still need to do:

- [ ] Update PyPI's metadata handling to match PEP 753 (the current behavior is virtually identical, but there are some edge cases + I'll add appropriate tests)
- [ ] Update PyPI's documentation to reference PEP 753
- [ ] Other, higher-level guides/docs that should be updated?
- [ ] Update the PEP itself to point to the canonical living PyPA spec

CC @pfmoore @ncoghlan @di as interested parties 🙂 

----
📚 Documentation preview 📚: https://python-packaging-user-guide--1611.org.readthedocs.build/en/1611/

### CODE REVIEW

File: source/specifications/well-known-project-urls.rst [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    maintainers) may continue to use any labels they please, within the
+    overall ``Project-URL`` length restrictions. However, users are
+    *encouraged* to pick meaningful labels that normalize to well-known
+    labels.
+
+.. note::
+
+    See :ref:`Writing your pyproject.toml - urls <writing-pyproject-toml-urls>`
+    for user-oriented guidance on choosing project URL labels in your package's
+    metadata.
```
Comment (pfmoore) [inline source/specifications/well-known-project-urls.rst:24]: This oversells the referenced section a bit. There's not really much "guidance" there, TBH.

**Review by pfmoore — APPROVED** [review]
A couple of minor comments, but overall LGTM!

File: source/specifications/well-known-project-urls.rst (hunk 1 above)
Comment (woodruffw) [reply source/specifications/well-known-project-urls.rst:24]: Fair! I'll mute it a bit.

Comment (pfmoore) [reply source/specifications/well-known-project-urls.rst]: ... or add a bit more to the referenced section? Maybe say there that users are advised to use well known labels where appropriate, as they will be rendered by consumers. And add a specific note about how to specify the project homepage and download URL. It doesn't have to be much, just sufficient to qualify as "guidance" :-)

Comment (woodruffw) [reply source/specifications/well-known-project-urls.rst:24]: Done, PTAL!

**Comment by woodruffw** [comment]
CI failure looks unrelated (linkcheck seems to be failing due to the current Internet Archive outage).

File: source/specifications/well-known-project-urls.rst [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+.. important::
+
+    This document is primarily of interest to metadata *consumers*,
+    who should use the normalization rules and well-known list below
+    to make their presentation of project URLs consistent across the
+    Python ecosystem.
+
+    Metadata *producers* (such as build tools and individual package
+    maintainers) may continue to use any labels they please, within the
+    overall ``Project-URL`` length restrictions. However, users are
```
Comment (willingc) [inline source/specifications/well-known-project-urls.rst:16]:
```suggestion
    overall ``Project-URL`` length restrictions. However, when possible, users are
```

File: source/specifications/well-known-project-urls.rst [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+:pep:`753` deprecates the :ref:`core-metadata-home-page` and
+:ref:`core-metadata-download-url` metadata fields in favor of
+:ref:`core-metadata-project-url`, and defines a normalization and
+lookup procedure for determining whether a ``Project-URL`` is
+"well-known," i.e. has the semantics assigned to ``Home-page``,
+``Download-URL``, or other common project URLs.
+
+This allows indices (such as the Python Package Index) and other downstream
+metadata consumers to specialize their presentations of project URLs in a
+consistent manner.
```
Comment (willingc) [inline source/specifications/well-known-project-urls.rst:37]:
```suggestion
metadata consumers to present project URLs in a
consistent manner.
```

File: source/specifications/well-known-project-urls.rst [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+Label normalization
+===================
+
+.. note::
+
+    Label normalization is performed by metadata *consumers*, not metadata
+    producers.
+
+To determine whether a ``Project-URL`` label is "well-known," metadata
+consumers should first normalize the label before comparing it to the
```
Comment (willingc) [inline source/specifications/well-known-project-urls.rst:50]:
```suggestion
consumers should normalize the label before comparing it to the
```

**Review by willingc — APPROVED** [review]
A few small edits. Thanks @woodruffw.

**Comment by woodruffw** [comment]
CI appears to have righted itself, so this should be good to go!

**Comment by pfmoore** [comment]
You have 2 approvals, I think this is ready to go. I can hit the merge button, apparently, but I'm not really an active committer to this repo, so I think it should be one of the guide editors who merges it.

**Review by webknjaz — APPROVED** [review]
Thanks!
